### PR TITLE
Update speech options to use `credentials`

### LINKED ISSUE #2459: [Tracking] Update webSpeechPonyFillFactory signature to accept a promise for region
This issue is to track the need to update the `webSpeechPonyFillFactory` signature so that we can provide a promise that returns both the `region` and `authorizationToken` together.

This will be necessary once the BF speech service resources are scaled up to more datacenters outside of `westus`. Currently, the region is hardcoded in the Emulator.

See PR https://github.com/microsoft/BotFramework-Emulator/pull/1878/files for context.

### PR DESCRIPTION
**Pull request by compulim**
> Fixes #2459.

### Fixed

-  Fixes [#2459](https://github.com/microsoft/BotFramework-WebChat/issues/2459). Updated Cognitive Services Speech Services to use latest fetch credentials signature, by [@compulim](https://github.com/compulim) in PR [#2740](https://github.com/microsoft/BotFramework-WebChat/pull/2759)

## Description

Updated to use `credentials` from [`web-speech-cognitive-services`](https://npmjs.com/package/web-speech-cognitive-services). Deprecating `authorizationToken`, `region`, and `subscriptionKey`.

## Specific Changes

- Update `createCognitiveServicesSpeechServicesPonyfillFactory` to use `credentials` in favor of `authorizationToken`, `region`, and `subscriptionKey`
- Add deprecation notes to outdated options
- Update all samples to use `credentials`

---

-  [x] Testing Added
   -  We currently do not have test for the ponyfill factory. We should port the test harness from DLSpeech SDK.

### CODE REVIEW

**Comment by compulim** [comment]
@tonyanziano this is the PR for having both `authorizationToken` and `region` in a single async function.

File: packages/bundle/src/createCognitiveServicesSpeechServicesPonyfillFactory.js [hunk 1]
```diff
@@ -12,6 +17,26 @@ export default function createCognitiveServicesSpeechServicesPonyfillFactory({
   subscriptionKey,
   textNormalization
 }) {
+  if (!credentials && (authorizationToken || region || subscriptionKey)) {
+    console.warn(
+      'botframework-webchat: "authorizationToken", "region", and "subscriptionKey" are being deprecated and will be removed on or after 2019-12-17. Please use "credentials" instead.'
```
Comment (corinagum) [inline packages/bundle/src/createCognitiveServicesSpeechServicesPonyfillFactory.js]:
```suggestion
      'botframework-webchat: "authorizationToken", "region", and "subscriptionKey" are being deprecated and will be removed on or after 2020-12-17. Please use "credentials" instead.'
```

**Comment by corinagum** [comment]
When I try to run the samples locally, I get the following error: 
image


getting started full bundle sample does not have the same error

File: packages/bundle/src/createCognitiveServicesSpeechServicesPonyfillFactory.js [hunk 2]
```diff
@@ -12,6 +17,26 @@ export default function createCognitiveServicesSpeechServicesPonyfillFactory({
...
+    );
+
+    credentials = async () => {
+      if (authorizationToken) {
```
Comment (corinagum) [inline packages/bundle/src/createCognitiveServicesSpeechServicesPonyfillFactory.js]: Doesn't this need to come from `credentials.authorizationToken`? I think I'm missing something.

**Comment by compulim** [comment]
I tried both `01/a.full-bundle` and `03.speech/b.cognitive-speech-services-js` loading locally-built `webchat.js` and it looks good to me on Chromium Edge. Probably fixed.

image

**Comment by rvallireply** [comment]
Hello, since a few hours we are getting an error using webchat with cognitive speech. 
Maybe related to this commit?
See https://github.com/microsoft/BotFramework-WebChat/issues/2802#issuecomment-574498604

**Comment by compulim** [comment]
@rvallireply for #2802, it is unrelated to this commit. It is a service-side issue.

For this PR, we do found a bug #2822 and is going to fix it in #2824.

If you see further issues, could you open a new issue? We might miss your comment in this PR because it is already merged and closed.

**Pull request merged**